### ISSUE TITLE
Quality issues in the plugin docs (incorrect asciidoc formatting, missing content)

### ISSUE THREAD

**Issue by dedemorton** [elastic/logstash#6693]
Ran across the following quality issues when I was scanning the input plugin docs for docgen issues. 

**Global:**

- [ ] Style of the description varies from plugin to plugin. Would be nice to make the descriptions for consistent in style (at least for the plugins that we maintain).

**Input plugins:**

- [x] In [plugin-inputs-beats](https://www.elastic.co/guide/en/logstash/5.2/plugins-inputs-beats.html), text overruns navigation.
- [ ] In [plugins-inputs-cloudwatch](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-cloudwatch.html):
  - [x] **Only 1 of the 3 examples is properly formatted as JSON.**
  - [x] "Configuration Example" heading is formatted incorrectly
  - [ ] "Configuration Example" should be sentence case and plural
  - [ ] List under `filter` option needs returns to render correct.  Correct typos.
- [x] **In [plugins-inputs-couchdb_changes](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-couchdb_changes.html), the intro has a couple of section titles that are incorrectly formatted.**
- [ ] Edit [plugins-inputs-file](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-file.html)
- [ ] **In [plugins-inputs-gelf](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-gelf.html), the example is not correctly formatted as xml.**
- [ ]  In [plugins-inputs-gemfire](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-gemfire.html):
- **Note is not formatted (needs to say NOTE: instead of Note: to get the correct formatting).**
- **Example under cache_xml_file is not correctly formatted as XML.**
- [ ] ⭐️ In [plugins-inputs-http_poller](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-http_poller.html):
- **Description of `validate_after_inactivity` option contains weird formatting and # sign.**
- This topic needs an edit. Contains typos, missing periods, etc.
- See https://github.com/logstash-plugins/logstash-input-http_poller/issues/99 
- [ ] In [plugins-inputs-imap](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-imap.html), lots of options are missing descriptions.
- [x] In multiple plugins, the example shown under the `id` option is incorrectly formatted. This must be boiler plate text that's reused from somewhere.
- [x] In [plugins-inputs-jmx](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-jmx.html), `## The configuration:` should be formatted as a subheading. (Fix configuration heading #46) 
- [x] In [plugins-inputs-kafka](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-kafka.html):
- **The contents of the `decorate_events` description should be formatted as a list.**
- https://www.elastic.co/guide/en/logstash/current/plugins-inputs-kafka.html#plugins-inputs-kafka-partition_assignment_strategy maps to kafka's `partition.assignment.strategy` property. (Rob/Chad-6-4-18)
- [ ] **In [plugins-inputs-lumberjack](https://www.elastic.co/guide/en/logstash/6.2/plugins-inputs-lumberjack.html), URL is formatted in markup style, not asciidoc.**
- [x] In [plugins-inputs-pipe](https://www.elastic.co/guide/en/logstash/5.2/plugins-inputs-pipe.html), description of the `command` option includes text that begins `TODO(sissel):`.
- [x] REMOVED [plugins-inputs-rackspace](https://www.elastic.co/guide/en/logstash/5.2/plugins-inputs-rackspace.html) contains no intro description.
- REMOVED [plugins-inputs-zenoss](https://www.elastic.co/guide/en/logstash/5.2/plugins-inputs-zenoss.html) contains no intro description.

**Output Plugins:**

- [ ] [plugins-outputs-boundary](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-boundary.html):
- needs an edit. What do people have against periods anyhow? Some of the descriptions are hard to follow.
- [x] In multiple plugins, the description of `workers` says, "TODO remove this in Logstash 6.0 when we no longer support the :legacy type This is hacky, but it can only be herne"
- [ ] [plugins-outputs-csv](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-csv.html) needs an edit.
- [ ] [plugins-outputs-datadog](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-datadog.html):
- Needs an edit. 
- Fixed: The intro is in the wrong place, so it's not being pulled into the topic. 
- Descriptions of the available options aren't very useful.
- [x] In [plugins-outputs-elasticsearch](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-elasticsearch.html), we point to an [old blog from 2013](https://www.elastic.co/blog/elasticsearch-versioning-support.). Not sure we should be doing that. Someone should at least confirm that the blog is still accurate. 
- [ ] In [plugins-outputs-elasticsearch](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html), add a note about migration from `doc` to `_doc`
- [x] In [plugins-outputs-google_bigquery](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-google_bigquery.html):
- Intro needs asciidoc tagging (sections plus examples properly tagged).
- [x] [plugins-outputs-google_cloud_storage](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-google_cloud_storage.html) has the same issues as above.
- [ ] [plugins-outputs-graphtastic](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-graphtastic.html) needs an edit. The writing seems almost confessional.
- [x] In [plugins-outputs-http](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-http.html), the intro description is missing because it's in the wrong place in the source file.
- [ ] In [plugins-outputs-kafka](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-kafka.html)
  - Fixed: text in table overruns navigation. 
  - **Formatting of options described under `acks` could be better.**
- [x] In [plugins-outputs-librato](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-librato.html), the intro description is missing because it's in the wrong place in the source file.
- [x] [plugins-outputs-lumberjack](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-lumberjack.html) contains no intro description.
- [x] [plugins-outputs-mongodb](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-mongodb.html) contains no intro description.
- [ ] **In [plugins-outputs-nagios_nsca](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-nagios_nsca.html), the example is not correctly formatted (gets broken into 2 examples).**
- [x] REMOVED [plugins-outputs-rackspace](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-rackspace.html) contains no intro description.
- [ ] [plugins-outputs-redmine](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-redmine.html) needs an edit. **Cringe-worthy typos.**
- [ ] **In [plugins-outputs-riemann](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-riemann.html), looks like the example coding might be off.**
- [x] **[plugins-outputs-s3](https://www.elastic.co/guide/en/logstash/6.2/plugins-outputs-s3.html) contains multiple formatting issues (incorrectly formatted bullet lists, headings, etc).**
- [x] [plugins-outputs-stomp](https://www.elastic.co/guide/en/logstash/5.2/plugins-outputs-stomp.html) contains no intro description.

**Filter plugins:**

- [ ] In [filters-elapsed](https://www.elastic.co/guide/en/logstash/current/plugins-filters-elapsed.html), correct default coding for `periodic_flush`. See discussion in [#36](https://github.com/logstash-plugins/logstash-filter-elapsed/pull/36)
- [x] In [plugins-filters-cipher](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-cipher.html), formatting for default value and value type is messed up (not sure if this is an asciidoc issue or the doc gen).
- [x] Bug in [plugins-filters-dns](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-dns.html)--edit link takes you to dissect.rb source.
- [x] Bug in [plugins-filters-dissect](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-dns.html)--edit link takes you to dns.rb source.
- [x] In [plugins-filters-dns.html](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-dns.html), the description of the `reverse` option includes to-do items: `TODO(sissel): The timeout limitation does seem to be fixed in here: http://redmine.ruby-lang.org/issues/5100 # but isn’t currently in JRuby. TODO(sissel): make action required? This was always the intent, but it due to a typo it was never enforced. Thus the default behavior in past versions was append by accident. Reverse resolve one or more fields.`
- [x] In [plugins-filters-dissect](https://www.elastic.co/guide/en/logstash/6.2/plugins-filters-dissect.html), some of the examples don't use the correct asciidoc example formatting type for the data that's shown. The docs are also very hard to follow. This is an important plugin.
- [ ] **In [plugins-filters-drop](https://www.elastic.co/guide/en/logstash/6.2/plugins-filters-drop.html), some of the examples don't use the correct asciidoc example formatting type for the data that's shown.**
- [ ]  In [plugins-filters-mutate](https://www.elastic.co/guide/en/logstash/current/plugins-filters-mutate.html), add an example to illustrate nesting 
- [ ] **In [plugins-filters-elapsed](https://www.elastic.co/guide/en/logstash/6.2/plugins-filters-elapsed.html), some of the examples don't use the correct asciidoc example formatting type for the data that's shown.**
- [ ] **In [plugins-filters-environment](https://www.elastic.co/guide/en/logstash/6.2/plugins-filters-environment.html), some of the examples don't use the correct asciidoc example formatting type for the data that's shown.**
- [x] REMOVED [plugins-filters-metaevent](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-metaevent.html) contains no intro description.
- [ ] **In [plugins-filters-metricize](https://www.elastic.co/guide/en/logstash/6.2/plugins-filters-metricize.html), the examples in the intro don't use the correct example formatting.**
- [x] REMOVEd In [plugins-filters-oui](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-oui.html), for example under the `source` option needs to be formatted.
- [ ] **In [plugins-filters-throttle](https://www.elastic.co/guide/en/logstash/6.2/plugins-filters-throttle.html).  Has been removed from CI.
  - [ ] some of the example formatting is messed up with the example being broken into a mixture of example and text.**
- [x] REMOVED In [plugins-filters-zeromq](https://www.elastic.co/guide/en/logstash/5.2/plugins-filters-zeromq.html), the bullet list in the intro isn't formatted correctly.

**Codec plugins:** 

- [ ] **In [plugins-codecs-cef](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-cef.html), the example under the `delimiter` option needs to be formatted correctly**
- [x] REMOVED [plugins-codecs-cloudtrail](https://www.elastic.co/guide/en/logstash/5.2/plugins-codecs-cloudtrail.html) doesn't really contain any doc content.
- [x] REMOVED [plugins-codecs-compress_spooler](https://www.elastic.co/guide/en/logstash/5.2/plugins-codecs-compress_spooler.html) does not contain an intro description. In fact, there's very little doc here. There is a description for `min_flush_time` in the Ruby source, but it doesn't appear in the docs (maybe it's newish??)
- [ ] [plugins-codecs-dots](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-dots.html) doesn't really contain any doc content. Needs an example at least.
- [ ] [plugins-codecs-edn](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-edn.html) doesn't really contain any doc content. Added description, but still needs an example.
- [ ] [plugins-codecs-edn_lines](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-edn_lines.html) doesn't really contain any doc content. Added description, but still needs an example
- [ ] [plugins-codecs-msgpack](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-msgpack.html) doesn't really contain any doc content. Added description, but still needs an example.
- [ ] **In [plugins-codecs-nmap](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-nmap.html), the list of event types should be formatted as a list.**
- [ ] **In [plugins-codecs-protobuf](https://www.elastic.co/guide/en/logstash/6.2/plugins-codecs-protobuf.html), link is formatted incorrectly and bullet list under the `include_path` option needs formatting.**

**Meta:** 
- [ ] In [filters-elapsed](https://www.elastic.co/guide/en/logstash/current/plugins-filters-elapsed.html), correct default coding for `periodic_flush`. See discussion in [#36](https://github.com/logstash-plugins/logstash-filter-elapsed/pull/36). **NOTE:** From CS: "I just looked at elastic/logstash#6693 and note that this also impacts the metrics, aggregate, multilines and throttle filters (per my quick search ^^)"

**Comment by dedemorton** [comment]
@karenzone This issue has been open for quite awhile because I use it to track and fix little nits that I notice in the plugin docs. Whenever you need to update the docs for a plugin, make sure you check this issue to see if there are any other fixes required.

**Comment by karenzone** [comment]
Most of the issues have been addressed. Moving unchecked issues to #10838 for research and remediation.  

Thanks for creating this meta issue and model for tracking quality issues in the plugin docs, DeDe. The good work you started lives on in #10838.
